src/webots/vrml/WbProtoManager.cpp

```diff
@@ -132,29 +132,31 @@ WbProtoModel *WbProtoManager::findModel(const QString &modelName, const QString
   if (protoDeclaration.isEmpty()) {
     // if no declaration was provided, attempt to find a valid one using the backwards compatibility mechanism
     protoDeclaration = injectDeclarationByBackwardsCompatibility(modelName);
+    const QString backwardsCompatibilityMessage =
+      tr("Please adapt your project to R2022b following these instructions: "
+         "https://github.com/cyberbotics/webots/wiki/How-to-adapt-your-world-or-PROTO-to-Webots-R2022b");
+    const QString outdatedProtoMessage =
+      tr("'%1' must be converted because EXTERNPROTO declarations are missing.").arg(parentFilePath);
+    const QString errorMessage =
+      tr("Missing declaration for '%1': Add 'EXTERNPROTO \"%2\"' in '%3'.")
+        .arg(modelName)
+        .arg(protoDeclaration.isEmpty() ? mWebotsProtoList.value(modelName)->url() : protoDeclaration)
+        .arg(parentFilePath);
```

Comment:
I think we can lower the scope of the messages to where they are actually used, no need to define them all when we only show some of them